nepalingo-web/src/hooks/StreakContext.tsx

```diff
@@ -38,52 +36,58 @@ export const StreakProvider = ({ children }: { children: ReactNode }) => {
       if (error && error.code !== "PGRST116") {
         console.error("Error fetching streak data:", error);
       }
+      return data;
+    }
+  };
 
-      if (data) {
-        setStreakStartDate(data.streak_start_date);
-        setCurrentStreak(data.current_streak);
-        setLongestStreak(data.longest_streak);
-        setStreakEndDate(data.streak_end_date);
-      }
+  const updateStreakState = async () => {
+    const data = await fetchStreakData();
+
+    if (data) {
+      setCurrentStreak(data.current_streak);
+      setLongestStreak(data.longest_streak);
     }
   };
 
   useEffect(() => {
-    fetchStreakData();
+    updateStreakState();
```

Comment:
Checking if the user exists would prevent any request if the user hasn't logged in. As the context is provided globally.